numpy/lib/_histograms_impl.py

```diff
@@ -261,8 +262,11 @@ def _hist_bin_auto(x, range):
     """
     fd_bw = _hist_bin_fd(x, range)
     sturges_bw = _hist_bin_sturges(x, range)
-    del range  # unused
-    if fd_bw:
+
+    # heuristic to limit the maximal number of bins
+    maximum_number_of_bins = 2 * x.size / math.log(x.size + 1)
```

Comment:
might as well 🤷🏻 

```suggestion
    maximum_number_of_bins = 2 * x.size / math.log1p(x.size)
```

Reply:
Cool! Never knew people needed it so much to add a special method for it